pybb/tests.py

```diff
@@ -151,6 +151,48 @@ def test_topic_addition(self):
         self.assertEqual(response.status_code, 200)
         self.assertTrue(Topic.objects.filter(name='new topic name').exists())
 
+    def test_topic_read_before_post_addition(self):
+        """
+        Test if everything is okay when :
+            - user A create the topic
+            - but before associated post is created, user B display the forum
+        """
+        topic = Topic(name='xtopic', forum=self.forum, user=self.user)
+        topic.save()
+        #topic is saved, but post is not yet created at this time
+
+        #an other user is displaing the forum before the post creation
+        user_ann = User.objects.create_user('ann', 'ann@localhost', 'ann')
+        client = Client()
+        client.login(username='ann', password='ann')
+
+        self.assertEqual(client.get(topic.get_absolute_url()).status_code, 404)
+        self.assertEqual(topic.forum.post_count, 1)
+        self.assertEqual(topic.forum.topic_count, 1)
+        #do we need to correct this ?
+        #self.assertEqual(topic.forum.topics.count(), 1)
+        self.assertEqual(topic.post_count, 0)
+        
+        #Now, TopicReadTracker is not created because the topic detail view raise a 404
+        #If its creation is not finished. So we create it manually to add a test, just in case
+        #we have an other way where TopicReadTracker could be set for a not complete topic.
+        TopicReadTracker.objects.create(user=user_ann, topic=topic, time_stamp=topic.created)
+        
+        #before correction, raised TypeError: can't compare datetime.datetime to NoneType
+        pybb_topic_unread([topic,], user_ann)
```

Comment:
If we are in test environment we can add some assertion there and for next line too.